core/market/src/db/dao/agreement.rs

```diff
@@ -58,9 +59,66 @@ pub enum AgreementDaoError {
     EventError(String),
     #[error("Invalid Agreement id: {0}")]
     InvalidId(#[from] ProposalIdParseError),
+    #[error("Agreement to client agreement error: {0}")]
+    ClientAgreement(ClientAgreementError),
 }
 
 impl<'c> AgreementDao<'c> {
+    pub async fn list(
+        &self,
+        node_id: Option<NodeId>,
+        state: Option<AgreementState>,
+        before: Option<NaiveDateTime>,
+        after: Option<NaiveDateTime>,
+        app_session_id: Option<String>,
+    ) -> Result<Vec<AgreementListEntry>, AgreementDaoError> {
```

Comment:
This is database level. I would list Agreements here, without converting to `AgreementListEntry`.
Current solution has a few drawbacks:
- On database level you would like to have as general functions as possible. Here you limit yourself to information that `AgreementListEntry` contains
- You anyway have to query whole Agreements from database, so there is no gain from database processing and filtering. The situation would be different, if you would query only specific fields
- You have to convert to `ClientAgreement` on database level. But in whole rest of the Market you use database Agreement. So this function won't be useful in any other context.
- You had to add `AgreementDaoError::ClientAgreement` error, which doesn't belong here, because it is not database error.
- Having `ClientAgreement` here is against market design, because conversion to client structures happens only on most external REST level or GSB level

Reply:
I removed the use of `ClientAgreement` entirely and now directly construct `AgreementListEntry` from the database agreement in REST and GSB endpoints.

Does this part look OK now?

Reply:
It's ok